server/lib/src/plugins/write_ops_counter.rs

```diff
@@ -0,0 +1,207 @@
+
+use kanidm_proto::internal::OperationError;
+
+use crate::event::{CreateEvent, DeleteEvent};
+
+use crate::prelude::ModifyEvent;
+use crate::prelude::Uuid;
+
+use super::{BatchModifyEvent, EntrySealedCommitted, Plugin, QueryServerWriteTransaction};
+
+pub struct WriteOperationCounter {}
+
+impl WriteOperationCounter {
+    fn increase_write_ops_counter(qs: &mut QueryServerWriteTransaction) {
+        qs.increase_write_ops_since_last_repl();
+    }
+
+    fn reset_write_ops_counter(qs: &mut QueryServerWriteTransaction) {
+        qs.reset_write_ops_since_last_repl();
+    }
+}
+
+impl Plugin for WriteOperationCounter {
+    fn id() -> &'static str {
+        "plugin_write_operation_counter"
+    }
+
+    fn post_create(
+        qs: &mut QueryServerWriteTransaction,
+        // List of what we committed that was valid?
+        _cand: &[EntrySealedCommitted],
+        _ce: &CreateEvent,
+    ) -> Result<(), OperationError> {
+        Self::increase_write_ops_counter(qs);
+        Ok(())
+    }
+    fn post_modify(
+        qs: &mut QueryServerWriteTransaction,
+        // List of what we modified that was valid?
+        _pre_cand: &[Arc<EntrySealedCommitted>],
+        _cand: &[EntrySealedCommitted],
+        _ce: &ModifyEvent,
+    ) -> Result<(), OperationError> {
+        Self::increase_write_ops_counter(qs);
+        Ok(())
+    }
+
+    fn post_batch_modify(
+        qs: &mut QueryServerWriteTransaction,
+        // List of what we modified that was valid?
+        _pre_cand: &[Arc<EntrySealedCommitted>],
+        _cand: &[EntrySealedCommitted],
+        _me: &BatchModifyEvent,
+    ) -> Result<(), OperationError> {
+        Self::increase_write_ops_counter(qs);
+        Ok(())
+    }
+
+    fn post_delete(
+        qs: &mut QueryServerWriteTransaction,
+        // List of what we delete that was valid?
+        _cand: &[EntrySealedCommitted],
+        _ce: &DeleteEvent,
+    ) -> Result<(), OperationError> {
+        Self::increase_write_ops_counter(qs);
+        Ok(())
+    }
+
+    fn post_repl_incremental(
+        qs: &mut QueryServerWriteTransaction,
+        _pre_cand: &[Arc<EntrySealedCommitted>],
+        _cand: &[EntrySealedCommitted],
+        _conflict_uuids: &BTreeSet<Uuid>,
+    ) -> Result<(), OperationError> {
+        Self::reset_write_ops_counter(qs);
+        Ok(())
+    }
+
+    fn post_repl_incremental_conflict(
+        qs: &mut QueryServerWriteTransaction,
+        _cand: &[(EntrySealedCommitted, Arc<EntrySealedCommitted>)],
+        _conflict_uuids: &mut BTreeSet<Uuid>,
+    ) -> Result<(), OperationError> {
+        Self::reset_write_ops_counter(qs);
+        Ok(())
+    }
+
+    fn post_repl_refresh(
+        qs: &mut QueryServerWriteTransaction,
+        _cand: &[EntrySealedCommitted],
+    ) -> Result<(), OperationError> {
+        Self::reset_write_ops_counter(qs);
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::prelude::*;
+
+    #[qs_test(domain_level=DOMAIN_LEVEL_7)]
```

Comment:
Don't need the domain_level set here.

Reply:
I must admit I blindly copied & pasted from the other tests, not having a clue how the macro works :D